cluster/aws/util.sh

```diff
@@ -36,7 +36,9 @@ source "${KUBE_ROOT}/cluster/aws/${KUBE_CONFIG_FILE-"config-default.sh"}"
 source "${KUBE_ROOT}/cluster/common.sh"
 source "${KUBE_ROOT}/cluster/lib/util.sh"
 
-ALLOCATE_NODE_CIDRS=true
+if [[ "${NETWORK_PROVIDER:-}" != "cni" ]]; then
```

Comment:
It doesn't need to be a Kubernetes specific CNI plugin - bridge + host-local is an example that would rely on cloud routes being configured.

Flannel does have an integration with Kubernetes where it expects the podCIDR field to be populated, and uses CNI.  See [kube-flannel](https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel.yml) + [code](https://github.com/coreos/flannel/blob/5dde68dbb9df5318768ab68d42fd055a94547f88/subnet/kube/kube.go).  It won't strictly need the cloud routes themselves, but does depend on the node cidrs being allocated.

As mentioned in another comment, Calico CNI also has an option to operate in a mode where it expects the podCIDR to exist and cloud routes to be configured.